File: frontend/src/types/deployment.ts
import type { Application } from './application';

export interface Deployment {
  id: string;
  application_id: string;
  version: string;
  status: 'Pending' | 'Deploying' | 'Success' | 'Failed';
  commit_hash: string;
  environment: 'dev' | 'staging' | 'production';
  deployed_at: string;
  logs_url: string;
  duration: number;
  created: string;
  updated: string;
  description: string;
  triggered_by: string;
  rollback_version: string;
  deployment_strategy: string;
  replicas: number;
  resources: {
    cpu: {
      request: string;
      limit: string;
    };
    memory: {
      request: string;
      limit: string;
    };
  };
  // Additional fields for UI
  application?: Application;
} 

File: backend/models/deployment.py
from pydantic import BaseModel
from typing import Optional, Dict, Any
from datetime import datetime

class DeploymentResources(BaseModel):
    cpu: Dict[str, str]
    memory: Dict[str, str]

class DeploymentCreate(BaseModel):
    application_id: str
    version: str
    environment: str
    commit_hash: str
    description: Optional[str] = ""
    triggered_by: Optional[str] = "admin@company.com"
    deployment_strategy: Optional[str] = "rolling"
    replicas: Optional[int] = 1
    resources: Optional[DeploymentResources] = None

class DeploymentUpdate(BaseModel):
    version: Optional[str] = None
    status: Optional[str] = None
    description: Optional[str] = None
    deployment_strategy: Optional[str] = None
    replicas: Optional[int] = None
    resources: Optional[DeploymentResources] = None

class Deployment(BaseModel):
    id: str
    application_id: str
    version: str
    status: str
    commit_hash: str
    environment: str
    deployed_at: str
    logs_url: str
    duration: int
    created: str
    updated: str
    description: str
    triggered_by: str
    rollback_version: str
    deployment_strategy: str
    replicas: int
    resources: DeploymentResources
    application: Optional[Dict[str, Any]] = None 